fuse/core.go

```diff
@@ -162,6 +162,18 @@ func (f *fuseNode) Attr(ctx context.Context, a *fuse.Attr) error {
 		return err
 	}
 	attr := plugin.Attributes(updatedEntry)
+	if !attr.HasSize() && plugin.ReadAction().IsSupportedOn(updatedEntry) {
+		// This Readable entry doesn't know its size, so download its content and set the size
+		// to that. This enables built-in tools like `cat` to work without slowing down `list`.
+		//
+		// TODO: How can we prevent tab-completion from being slow?
```

Comment:
Not sure how to do this. Could we check for tab-completion process by checking for the shell's PID?

Either way, I think moving this stuff over to FUSE is the right move. It seemed weird to include a size sometimes, and not include a size other times -- `plugin.Attributes` should return what the plugin author provided.

Reply:
Also OK with deferring the fix for later since other users have not yet raised this issue.

Reply:
Yeah, checking for the shell's PID might actually work.

Reply:
Good news is this fix doesn't slow down `ls` or `find` since those use the API. Right now it's just tab-completion.